### DISCUSSION TITLE
[2021.09.14] 떡볶이 떡 만들기

### DISCUSSION THREAD

**Discussion by pyunji** [pyunji/Hyundai-ITnE-Team_QuadCore#131] [Algorithm]
# 문제: 떡볶이 떡 만들기
### 문제 설명
- 오늘 동빈이는 여행 가신 부모님을 대신해서 떡집 일을 하기로 했습니다. 오늘은 떡볶이 떡을 만드는 날입니다. 동빈이네 떡볶이 떡은 재밌게도 떡볶이 떡의 길이가 일정하지 않습니다. 대신에 한 봉지 안에 들어가는 떡의 총 길이는 절단기로 잘라서 맞춰줍니다.
- 절단기에 **높이(H)** 를 지정하면 줄지어진 떡을 한 번에 절단합니다. 높이가 H보다 긴 떡은 H 위의 부분이 잘릴 것이고, 낮은 떡은 잘리지 않습니다.
- 예를 들어 높이가 19, 14, 10, 17cm인 떡이 나란히 있고 절단기 높이를 15cm로 지정하면 자른 뒤 떡의 높이는 15, 14, 10, 15가 될 것입니다. 잘린 떡의 길이는 차례대로 4, 0, 0, 2cm입니다. 손님은 6cm 만큼의 길이를 가져갑니다. 
- 손님이 왔을 때 요청한 총 길이가 M일 때 **적어도 M만큼의 떡을 얻기 위해 절단기에 설정할 수 있는 높이의 최댓값을 구하는 프로그램**을 작성하세요.

### 입력 조건
- 첫째 줄에 떡의 개수 N과 요청한 떡의 길이 M이 주어집니다. (1 <= N <= 1,000,000, 1<= M <= 2,000,000,000)
- 둘째 줄에는 떡의 개별 높이가 주어집니다. 떡 높이의 총합은 항상 M 이상이므로, 손님은 필요한 양만큼 떡을 사갈 수 있습니다. 높이는 10억보다 작거나 같은 양의 정수 또는 0입니다.

### 출력 조건
- 적어도 M만큼의 떡을 집에 가져가기 위해 절단기에 설정할 수 있는 높이의 최댓값을 출력합니다.

### 입력 예시
```
4 6
19 15 10 17
```

### 출력 예시
```
15
```

#### ❗[동영상 참고] (https://www.youtube.com/watch?v=94RC-DsGMLo)

**Comment by chaeyeon4u** [comment]
[떡볶이 떡 만들기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/Java/%EB%96%A1%EB%B3%B6%EC%9D%B4%EB%96%A1%EB%A7%8C%EB%93%A4%EA%B8%B0.java)

**Reply by chaeyeon4u** [reply]
[떡볶이떡 만들기](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/Java/%EC%9D%B4%EC%A7%84%ED%83%90%EC%83%89%20%EB%96%A1%EB%B3%B6%EC%9D%B4%EB%96%A1%202.java)

**Comment by hyk339** [comment]
[떡볶이 떡 만들기 solved](https://github.com/hyk339/Hyundai-ITnE-Team_QuadCore/commit/fdf3aa77a424431e96e7f14611452434dcbd097f)

**Comment by pyunji** [comment]
[떡볶이 떡 만들기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/pyunji/pyunji/Algorithm/Basic_BinarySearch2.java)

**Comment by leechang4** [comment]
[떡볶이 떡 만들기 solved](https://github.com/leechang4/Hyundai-ITnE-Team_QuadCore/blob/master/lch/DataStructure/Basic_BinarySearch2.java)
